File: SelectionSort.py
'''
Selection sort algorithm implementation in Python
Space complexity = O(1)
Time complexity = O(n^n)
'''
def selectionsort(arr):
    for i in range(len(arr)-1):
        ind=i

        # Finding min num in the unsorted subarray
        for j in range(i+1,len(arr)):
            if (arr[j]<arr[ind]):
                ind=j

        # Swap the arr numbers
        arr[i], arr[ind] = arr[ind], arr[i]

    return arr

# To print list of numbers as a comma separated string
print(", ".join(map(str,selectionsort([2,45,6,10,34,51,10,68,100,1]))))

File: SelectionSort.java
/*
Selection sort algorithm implementation in Java
Space complexity = O(1)
Time complexity = O(n*n)
*/
public class SelectionSort
{

	private static int[] sort(int arr[]){

		for (int i=0;i<arr.length-1 ;i++ )
		{
			int ind = i;
			for (int j=i+1;j<arr.length ;j++ )
			{
				//Finding minimum element
				if (arr[j]<arr[ind])
				{
					ind=j;
				}
			}

			//Swap the elements
			int temp = arr[ind];
			arr[ind] = arr[i];
			arr[i] = temp;
		}

		return arr;
	}

	public static void main(String args[]){
		int arr[] = {24,7,54,33,10,37,62,10,44,29};
		
		for (int x : sort(arr)){
			System.out.print(x+" ");
		}
	}
}
